modules/csvparser/csvparser.c

```diff
@@ -180,24 +206,10 @@ _iterate_columns(CSVParser *self, CSVScanner *scanner, LogMessage *msg)
     {
       if (self->columns)
         {
-          CSVParserColumn *current_column = column_l->data;
-          const gchar *current_value = csv_scanner_get_current_value(scanner);
-          GError *error = NULL;
-
-          if (self->drop_invalid && !type_cast_validate(current_value, current_column->type, &error))
+          if (!_process_column(self, scanner, msg, column_l, key_scratch, _key_formatter))
```

Comment:
we should not pass the GList iterator here, passing current_column seems better.

Reply:
the _key_formatter could be also pushed into the _process_column() function, removing yet another argument.